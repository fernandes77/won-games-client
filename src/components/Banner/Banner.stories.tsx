import type { Meta, StoryObj } from '@storybook/react'
import { Banner } from './Banner'

const meta: Meta<typeof Banner> = {
  component: Banner
}

export default meta
type Story = StoryObj<typeof Banner>

export const Primary: Story = {
  args: {
    buttonLabel: 'Buy now',
    buttonLink: '/games',
    img: 'https://source.unsplash.com/user/willianjusten/1042x580',
    subtitle: '<p>Play the new <strong>CrashLands</strong> season',
    title: 'Defy death'
  },
  parameters: { layout: 'fullscreen' },
  decorators: [
    (Story) => (
      <div style={{ maxWidth: '65rem', margin: '0 auto' }}>
        <Story />
      </div>
    )
  ]
}

export const WithRibbon: Story = {
  args: {
    buttonLabel: 'Buy now',
    buttonLink: '/games',
    img: 'https://source.unsplash.com/user/willianjusten/1042x580',
    subtitle: '<p>Play the new <strong>CrashLands</strong> season',
    title: 'Defy death',
    ribbon: '20% OFF',
    ribbonSize: 'md',
    ribbonColor: 'primary'
  },
  parameters: { layout: 'fullscreen' },
  decorators: [
    (Story) => (
      <div style={{ maxWidth: '65rem', margin: '0 auto' }}>
        <Story />
      </div>
    )
  ]
}
