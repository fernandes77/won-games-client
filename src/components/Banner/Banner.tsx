import { Button } from '@/components/Button/Button'

export type BannerProps = {
  img: string
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
}

export const Banner = ({
  img,
  title,
  subtitle,
  buttonLabel,
  buttonLink
}: BannerProps) => (
  <div className="relative md:shadow-[0_4px_5px_0_rgba(0,0,0,0.2)]">
    <div
      className={`w-full h-56 bg-gray-light bg-center bg-cover md:h-[36rem]`}
      role="img"
      aria-label={title}
      style={{ backgroundImage: `url(${img})` }}
    />

    <div className="w-full bg-[rgba(0, 0, 0, 0.7)] p-sm md:rounded-b md:p-lg md:absolute md:bottom-0 md:left-0">
      <h2 className="text-lg text-bold text-white md:text-xxl">{title}</h2>
      <h3
        className="text-white text-sm mb-xs md:text-lg [&_strong]:text-primary [&_strong]:text-bold"
        dangerouslySetInnerHTML={{ __html: subtitle }}
      />
      <Button as={{ type: 'link', props: { href: buttonLink } }} size="lg">
        {buttonLabel}
      </Button>
    </div>
  </div>
)
