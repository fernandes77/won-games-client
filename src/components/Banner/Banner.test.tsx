import { render, screen } from '@/utils/tests/helpers'
import { Banner } from './Banner'

const props = {
  img: 'https://source.unsplash.com/user/willianjusten/1042x580',
  title: 'Defy death',
  subtitle: '<p>Play the new <strong>CrashLands</strong> season',
  buttonLabel: 'Buy now',
  buttonLink: '/games/defy-death'
}

describe('<Banner />', () => {
  it('should render correctly', () => {
    const { container } = render(<Banner {...props} />)

    expect(
      screen.getByRole('heading', { name: /Defy death/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Play the new CrashLands season/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /Defy death/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a Ribbon', () => {
    render(
      <Banner
        {...props}
        ribbon="My Ribbon"
        ribbonSize="sm"
        ribbonColor="secondary"
      />
    )

    const ribbon = screen.getByText(/My Ribbon/i)

    expect(ribbon).toBeInTheDocument()
    expect(ribbon).toHaveClass('bg-secondary')
    expect(ribbon).toHaveClass('text-xs h-[1.625rem]')
  })
})
