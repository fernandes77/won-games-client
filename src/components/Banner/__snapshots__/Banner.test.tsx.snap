// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
<div
  class="relative md:shadow-[0_4px_5px_0_rgba(0,0,0,0.2)]"
>
  <div
    aria-label="Defy death"
    class="w-full h-56 bg-gray-light bg-center bg-cover md:h-[36rem]"
    role="img"
    style="background-image: url(https://source.unsplash.com/user/willianjusten/1042x580);"
  />
  <div
    class="w-full bg-[rgba(0, 0, 0, 0.7)] p-sm md:rounded-b md:p-lg md:absolute md:bottom-0 md:left-0"
  >
    <h2
      class="text-lg text-bold text-white md:text-xxl"
    >
      Defy death
    </h2>
    <h3
      class="text-white text-sm mb-xs md:text-lg [&_strong]:text-primary [&_strong]:text-bold"
    >
      <p>
        Play the new 
        <strong>
          CrashLands
        </strong>
         season
      </p>
    </h3>
    <a
      class="h-[3.125rem] text-md px-xl py-xxs inline-flex items-center justify-center cursor-pointer bg-gradient-to-t from-[#ff5f5f] via-[#f062c0] to-[#f062c0] text-white rounded p-xxs [&>svg]:w-4 hover:bg-gradient-to-t hover:from-[#e35565] hover:via-[#d958a6] hover:to-[#d958a6]"
      href="/games/defy-death"
    >
      <span
        class=""
      >
        Buy now
      </span>
    </a>
  </div>
</div>
`;
